### PR TITLE
nbc.py patch

### PR DESCRIPTION
**Pull request by 2ShedsJackson**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Sorry for the repost. My repository somehow got corrupted and I didn't want to waste anymore time troubleshooting it. Anyway,

The nbc.py extractor was using the wrong information for video titles. This was noticeable in movies: For example, "The Fifth Element" would end up with the title "1 hr 48 min _ 1997". The extractor was using the secondaryTitle key from one JSON file instead of the title key from a different JSON file. The two keys usually have the same value for episodes of TV shows.

age_limit information was added.

Fallbacks were added.

### CODE REVIEW

**Comment by 2ShedsJackson** [comment]
To recap the previous discussion, using the `or` conditional for a shortcut doesn't work because valid values can be `0`. The code has to check for `None` specifically. Also, one-liners that test for `None` will become unwieldy when values each have two or more fallbacks.

File: yt_dlp/extractor/nbc.py [hunk 1]
```diff
@@ -142,18 +145,40 @@ def _real_extract(self, url):
         theplatform_url = smuggle_url(update_url_query(
             'http://link.theplatform.com/s/NnzsPC/media/guid/%s/%s' % (video_data.get('mpxAccountId') or '2410887629', video_id),
             query), {'force_smil_url': True})
+
+        # fallbacks
```
Comment (pukkandan) [inline yt_dlp/extractor/nbc.py:149]: Add a comment here why `or` is intentionally not used; so that someone else doesn't try to refactor it in the future

```suggestion
        # Empty string or 0 can be valid values for these. So the check must be `is None`
```

File: yt_dlp/extractor/nbc.py [hunk 2]
```diff
@@ -142,18 +145,40 @@ def _real_extract(self, url):
...
+            try_get(tpm, lambda x: x['ratings'][0]['rating'])
+        season_number = int_or_none(video_data.get('seasonNumber'))
+        if season_number is None:
+            season_number = int_or_none(tpm.get('nbcu$seasonNumber'))
+        series = video_data.get('seriesShortTitle')
+        if series is None:
+            series = tpm.get('nbcu$seriesShortTitle')
+        tags = video_data.get('keywords')
+        if tags is None or len(tags) == 0:
+            tags = tpm.get('keywords')
```
Comment (pukkandan) [inline yt_dlp/extractor/nbc.py:167]: `series` and `tags` can be simplified and moved below, no?

**Comment by pukkandan** [comment]
I didn't know reviews disappear on forcepush 😢
